lib/rules/switch-final-break.js

```diff
@@ -0,0 +1,189 @@
+                if (requireBreak !== "always") {
+                    return;
+                }
+                if (node.type === "SwitchCase" && node.consequent.length === 0) {
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            node,
+                            " break;"
+                        )
+                    });
+                }
+                if (currentCodePath.currentSegments.some(isReachable) &&
+                    node.consequent.length > 0 &&
+                    lodash.last(node.parent.cases) === node
+                ) {
+                    const lastStatement = getLastStatement(node);
+
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => {
+                            const lastStatementText = context.getSourceCode().getLines()[
+                                lastStatement.loc.start.line - 1
+                            ];
+                            const indentation = lastStatementText.slice(
+                                0,
+                                lastStatementText.search(/\S+/u)
+                            );
+
+                            return fixer.insertTextAfter(
+                                lastStatement,
+                                `\n${indentation}break;`
+                            );
+                        }
+                    });
+                }
+            },
+
+            SwitchStatement(node) {
+                if (node.cases.length !== 0) {
+                    const lastCase = node.cases[node.cases.length - 1];
+                    const lastStatement = getLastStatement(lastCase);
+
+                    if (requireBreak === "always") {
+                        return;
+                    }
+
+                    if (
+                        !lastStatement ||
+                        lastStatement.type !== "BreakStatement"
+                    ) {
+                        return;
+                    }
+
+                    if (lastStatement.label !== null) {
+                        const parent = node.parent;
+
+                        if (
+                            parent &&
+                            (!(parent.type === "LabeledStatement") ||
+                                parent.label === lastStatement.label)) {
+
+                            // break jumps somewhere else, don't complain
+                            return;
+                        }
+                    }
+
+                    context.report({
+                        node: lastStatement,
+                        messageId: "switchFinalBreakNever",
+                        fix: fixer => {
+                            const sourceText = context.getSourceCode().getText();
```

Comment:
Is all this calculation necessary? Can we remove the `break` statement using the `start` and `end` properties of the `lastStatement` node?

Reply:
this is used to remove leftovers spaces and empty lines. Do you find it unnecessary?

Reply:
My preference would be to not do all this calculation, since I think we should keep fixers as small as possible and fine-tune as necessary if we get issues from folks.